llvm/lib/Support/DataExtractor.cpp

```diff
@@ -48,7 +48,7 @@ T DataExtractor::getU(uint64_t *offset_ptr, Error *Err) const {
   if (!prepareRead(offset, sizeof(T), Err))
     return val;
   std::memcpy(&val, &Data.data()[offset], sizeof(val));
-  if (sys::IsLittleEndianHost != IsLittleEndian)
+  if (!IsLittleEndian)
```

Comment:
The original code is correct.

This change causes llvm/unittests/Support/DataExtractorTest.cpp failures, but I am surprised that there are only a few failures.